hooks/use-autoclaim.ts

```diff
@@ -1,28 +1,25 @@
 import { useCallback, useEffect, useState } from "react";
 import { useAccount, useWriteContract, useWaitForTransactionReceipt } from "wagmi";
-import CONTRACTS from "@/utils/contracts";
+import CONTRACTS, { ContractNetwork } from "@/utils/contracts";
 import claimRegistryABI from "@/utils/abis/claimRegistry";
 import { getPublicClient } from "wagmi/actions";
 import { config } from "@/wagmi";
 import { fetchRegister, fetchUnregister } from "@/utils/fetchEvents";
 import { parseUnits } from "viem";
 
-function useAutoclaim() {
-  const account = useAccount();
+function useAutoclaim(contractConfig: ContractNetwork | undefined, address: `0x${string}` | undefined, chainId: number) {
   const [isRegister, setIsRegister] = useState(false);
-  const chainId = account?.chainId || 100;
-  const contractConfig = CONTRACTS[chainId];
-  const client = getPublicClient(config, { chainId: chainId as 100 | 10200 });
+  const client = getPublicClient(config, { chainId: chainId as 100 | 10200 | 31337 });
   const { data: autoclaimHash, writeContract } = useWriteContract();
   const { isSuccess: autoclaimSuccess } = useWaitForTransactionReceipt({
     hash: autoclaimHash,
   });
 
   useEffect(() => {
     async function fetchEvents() {
-      if (contractConfig && account.address && account.address !== "0x0") {
-        const registerEvents = await fetchRegister(contractConfig.addresses.claimRegistry, account.address, contractConfig.claimRegistryStartBlockNumber, client);
-        const unregisterEvents = await fetchUnregister(contractConfig.addresses.claimRegistry, account.address, contractConfig.claimRegistryStartBlockNumber, client);
+      if (contractConfig && address && address !== "0x0") {
+        const registerEvents = await fetchRegister(contractConfig.addresses.claimRegistry, address, contractConfig.claimRegistryStartBlockNumber, client);
+        const unregisterEvents = await fetchUnregister(contractConfig.addresses.claimRegistry, address, contractConfig.claimRegistryStartBlockNumber, client);
 
         if (registerEvents.length == 0 && unregisterEvents.length == 0) {
           setIsRegister(false);
```

Comment:
Not connected to this PR, but I think it's possible not to use events log (even no subgraph needed) and to read from the `claimRegistry` contract directly as it keeps status in the public `configs` mapping.

Here I've implemented this approach, while playing with envio:
https://github.com/gnosischain/gbc-deposit-ui/blob/3eab96bf1edb979aff9d1f5273d3cf5235b698fb/hooks/use-autoclaim.ts#L21C2-L30C5

Reply:
Nice improvement i agree! It seems more straightforward than the current implementation 👍